tools/gen-postmortem-metadata.py

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env python
+
+#
+# gen-postmortem-metadata.py output_file.cc
+#
+# Creates debugging symbols to help naviage Node's internals using post-mortem
+# debugging tools.
+#
+
+import sys
+
+
+class DebugSymbol(object):
+    type_ = 'int'
+    _PREFIX = 'nodedbg_'
+
+    def __init__(self, name, value, headers=[], type_=None):
+        self.name = name
+        self.value = value
+        self.headers = headers
+        self.type_ = type_ or DebugSymbol.type_
+
+    @classmethod
+    def get_headers(cls, debug_symbols):
+        seen = set()
+        seen_add = seen.add  # faster than seen.add on each iteration
+        headers = [debug_symbol.headers for debug_symbol in debug_symbols]
+        headers = sum(headers, [])
+
+        return [h for h in headers if not (h in seen or seen_add(h))]
+
+    def __str__(self):
+        return "{type} {prefix}{name} = {value};".format(
+            type=self.type_,
+            prefix=self._PREFIX,
+            name=self.name,
+            value=self.value,
+        )
+
+
+debug_symbols = [
+    DebugSymbol(
+        name="currentEnvironment",
+        value="(uint64_t) &Environment::currentEnvironment",
+        headers=["env.h"],
+        type_="uint64_t",
+    ),
+    DebugSymbol(
+        name="class__BaseObject__persistant_handle",
+        value="(size_t) &(((BaseObject*)0)->persistent())",
+        headers=["base-object.h", "base-object-inl.h"],
+        type_="size_t",
+    ),
+    DebugSymbol(
+        name="class__Environment__handleWrapQueue",
+        value="(size_t) (((Environment*)0)->handle_wrap_queue())",
```

Comment:
Tbh, I’d just put `#define private public` at the top of the generated file and use `offsetof`.